### PR TITLE
Feature/split testing

### LINKED ISSUE #166: Testing models
We need to test the higher level model classes, where should we do this? At the moment, a number of tests are combined in `test/test_models.py` but this might soon become overcrowded.

Should we split the file to test for each Model individually or keep it as one file? Some functionalities will be the same for all models, maybe tests could be combined.

### LINKED ISSUE #168: Restructure the `tests` folder.
The `tests` folder has a very flat structure, which is going to get confusing fast. @dalonsoa has pointed out:

> In complex projects, I normally keep for the tests the same folder/file structure than in the actual code. In that sense, I would restructure completely the location of the tests to follow that. I'm sorry I did not spot this back at the beginning, when starting to write the tests.

_Originally posted by @dalonsoa in https://github.com/ImperialCollegeLondon/virtual_rainforest/issues/166#issuecomment-1422476713_

### PR DESCRIPTION
**Pull request by jacobcook1995**
# Description

The test files have now been split such that their structure mirrors the structure of the package code. This PR also changes the name of the model specific `model.py` to `{MODEL_NAME}_model.py` to make things a bit clearer.

Fixes #166 
Fixed #168

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by davidorme** [comment]
In addition to restructuring the test files to match the code directory structure, this also reorganises the test data files, merging the `fixtures` folder into the `data` folder used by the `pytest-datadir` extension.

**Review by davidorme — APPROVED** [review]
LGTM

File: tests/core/test_model.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+"""Test module for model.py (and associated functionality).
+
+This module tests the functionality of model.py, as well as other bits of code that
+define models based on the class defined in model.py
```
Comment (vgro) [inline tests/core/test_model.py:4]: would you like to add here {MODEL_NAME}_model.py? Or mention that this is a general test for the model?

**Review by vgro — APPROVED** [review]
LGTM, one minor comment.

File: tests/core/test_model.py (hunk 1 above)
Comment (jacobcook1995) [reply tests/core/test_model.py]: Yeah that bit definitely could be clearer. I've tried to clarify that this script tests `model.py` which is the script that provides the framework that other specific models (e.g. `abiotic_model.py`) have to follow

**Pull request merged**